kms/managed.go

```diff
@@ -0,0 +1,216 @@
+package kms
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	// CollisionTolerance with 2^21 as a very defensive value. 2^32 is more commonly used.
+	CollisionTolerance = 2097151
+	// keySize is the key size in bytes
+	keySize = 128 / 8
+	// nonceSize is the size of the nonce. Do not change, without breaking version change.
+	// The nonceSize is a de facto standard.
+	nonceSize = 12
+	// cacheSize is set to 100 keys, which can be used to decrypt up to 200 million encrypted
+	// files or data encrypted within nearly 2 years.
+	cacheSize = 100
+)
+
+var (
+	// ErrKeyExpired means that the expiration time of a key has come and it shouldn't be used any more.
+	ErrKeyExpired = errors.New("key is out of date and shouldn't be used anymore for encryption")
+	// ErrNoCipher means that there is no upstream kms given and therefore the keys in use can't be protected.
+	ErrNoCipher = errors.New("no upstream encryption service was specified")
+
+	// expirationTime of a Week
+	expirationTime = time.Hour * 24 * 7
```

Comment:
if it's hardcoded to a week, then the var name should reflect that. e.g. `expirationTimeOneWeek`